### PR TITLE
changed some String.format into String template

### PR DESCRIPTION
**Pull request by abhi835**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
Fixes _Link to the issues._

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -22,7 +22,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.3.0'
+        classpath 'com.android.tools.build:gradle:7.4.0'
```
Comment (criticalAY) [inline build.gradle:25]: This gradle update isn't required in PR

File: README.md [hunk 2]
```diff
@@ -17,14 +17,23 @@
...
-
++ 
 ### Features
 
 <div style="display:flex;">
  
 - night mode
 - whiteboard 
 - progress widget
+
```
Comment (criticalAY) [inline README.md:28]: Unnecessary blank lines. Please revert the changes

**Review by criticalAY — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (mikehardy) [reply build.gradle:25]: Stated more strongly: never include unrelated items in a PR, *especially* changes to build files, those changes are always unwanted. Have a focused goal for the PR, then with the changes include only the minimal things required to implement the goal of the PR

**Review by abhi835 — COMMENTED** [review]
Changed to previous version

File: README.md [hunk 3]
```diff
@@ -17,7 +17,7 @@
 A semi-official port of the open source [Anki](https://apps.ankiweb.net/index.html) spaced repetition flashcard system to Android. Memorize anything with AnkiDroid!
 
 <img src="docs/graphics/logos/ankidroid_logo.png" align="right" width="40%" height="100%"></img>
-
+-
```
Comment (criticalAY) [inline README.md:20]: Revert this one too

**Review by criticalAY — CHANGES_REQUESTED** [review]
Looks good just need to revert one last change and we are good to go.

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.kt [hunk 4]
```diff
@@ -618,7 +618,8 @@ public class AddContentApi(context: Context) {
             // Loop through each item in fieldsArray looking for an existing note, and add it to the duplicates array
             val queryFormat = "${modelFieldList[0]}:\"%%s\" note:\"$modelName\""
             for (outputPos in keys.indices) {
-                val selection = String.format(queryFormat, keys[outputPos])
+                val selection = "$queryFormat${keys[outputPos]}"
+
```
Comment (n1snt) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.kt:622]: This line also needs to be removed.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 5]
```diff
@@ -550,9 +550,8 @@ class DeckOptions :
     }
 
     // TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
-    @KotlinCleanup("remove reduntant val res = resources")
+
```
Comment (n1snt) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:553]: This too

**Comment by n1snt** [comment]
There are some merge conflicts too that need to be resolved.

**Pull request closed without merging**